miniscope_io/cli/update.py

```diff
@@ -0,0 +1,77 @@
+"""
+CLI for updating device over IR or UART.
+"""
+
+import click
+import yaml
+
+from miniscope_io.device_update import DevUpdate
+from miniscope_io.models.devupdate import DeviceCommand
+
+
+@click.command()
+@click.option(
+    "-p",
+    "--port",
+    required=False,
+    help="Serial port to connect to. Needed if multiple FTDI devices are connected.",
+)
+@click.option(
+    "-i",
+    "--device_id",
+    required=False,
+    default=0,
+    type=int,
+    help="ID of the device to update. 0 will update all devices.",
+)
+@click.option(
+    "-t",
+    "--target",
+    required=False,
+    type=click.Choice(["LED", "GAIN", "ROI_X", "ROI_Y"]),
+    help="Target to update. Cannot be used with --restart.",
+)
+@click.option(
+    "-v",
+    "--value",
+    required=False,
+    type=int,
+    help="Value to set. Must be used with --target and cannot be used with --restart.",
+)
+@click.option(
+    "-c",
+    "--config",
+    required=False,
+    type=click.Path(exists=True, dir_okay=False),
+    help="YAML file with configuration to update. Specify target and value pairs in the file.",
```

Comment:
I separated this because the device config is static unless the firmware is updated, but the user will likely interactively change this update stuff while looking at the data, and having them together felt awkward.
It's just a lazy batch interface that literally sends the YAML file's keys/value pairs like `mio update --key KEY --value VALUE`.

This batch stuff isn't strictly necessary, and I'm not sure what's the best way to have this, so I'd like to keep it and just note this is an experimental feature in the help if that sounds ok.

Reply:
> the user will likely interactively change this update config stuff while looking at the data, and having them together felt awkward.

def true. to make them be the same thing would require us making this update function idempotent and good at filtering out irrelevant values and unchanged values. fine to keep as is, think the note would be helpful so that ppl know this is a different file than what we mean when we use "config" elsewhere (we might even just name it "yaml" to say "pass k/v pairs via yaml" rather than call it a "config" to avoid confusion, but idrc about that)

Reply:
This is basically a batch so I will switch to `yaml_batch`. Not beautiful so pretty temporary.